iot_config/src/gateway_service.rs

```diff
@@ -71,7 +72,12 @@ impl iot_config::Gateway for GatewayService {
             .await
             .map_err(|_| Status::internal("gateway lookup error"))?;
 
-        let location = location.ok_or_else(|| Status::internal("gateway location undefined"))?;
+        let location = {
+            let location =
+                location.ok_or_else(|| Status::internal("gateway location undefined"))?;
+            Cell::from_raw(location)
+                .map_err(|_| Status::permission_denied("invalid h3 index"))?
```

Comment:
not a fan of the fairly limited offering tonic `Status` options we have to provide feedback to the user; i have been leaning on `internal` errors whenever it's caused by a problem on "our" end which i would argue we have here since the internal mechanics of the oracles/chain are trying to retrieve the asserted location. in as far as anyone is looking at their gateway logs this feels like it's not actionable by the operator since they have no direct control over what h3 index the gateway is asserted against to correct this but if anything i'd argue it's at least an `invalid_argument` error. thought?

Reply:
I hadn't formed any mental image for this and your rationale seems sound. I didn't think about the fact that the user can't really do anything about the extremely unlikely chance their hotspot has an invalid index.